AdminWebsite/AdminWebsite/ClientApp/src/app/booking/search-email/search-email.component.ts

```diff
@@ -43,23 +48,37 @@ export class SearchEmailComponent implements OnInit, OnDestroy {
 
     ngOnInit() {
         this.$subscriptions.push(
-            this.searchService.search(this.searchTerm, this.hearingRoleParticipant).subscribe(data => {
-                if (data && data.length > 0) {
-                    this.getData(data);
-                } else {
-                    if (this.email.length > 2) {
-                        this.noDataFound();
+            this.searchTerm
+                .pipe(
+                    debounceTime(500),
+                    distinctUntilChanged(),
+                    tap(() => {
+                        this.searchPending.next(true);
+                    }),
+                    switchMap(term => {
+                        return this.searchService.participantSearch(term, this.hearingRoleParticipant);
+                    })
+                )
+
+                .subscribe(personsFound => {
+                    if (personsFound && personsFound.length > 0) {
+                        this.getData(personsFound);
```

Comment:
Is there a reason getData isn't just in the pipe?

Reply:
Not my change, I'll look into it.